samples/openthread/cli/Kconfig

```diff
@@ -4,6 +4,16 @@
 # SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
 #
 
+if NET_L2_OPENTHREAD
+
+config OPENTHREAD_LIBRARIES_SUBSTITUTE
+	bool "Substitute OpenThread Libraries"
+	depends on OPENTHREAD_SOURCES
+	help
+	  OpenThread libraries compiled from source will replace those provided by the nrfxlib.
```

Comment:
> Are end-users expected to replace the libs provided by nrfxlib ?
Yes, there are cases in which users may want to replace libs provided by nrfxlib.

I think the most important one is creating customized version that would be certified and used for other projects.
Provided mechanism makes it very easy.

Another benefit is for us, it makes libraries update process a piece of cake.

>Why do the end-user need to do this while compiling openthread CLI ?
Good question. My initial approach was to provide it as a completely separate example.
I was advised to to combine it with the CLI sample.

It makes sense as the CLI can be used to test pretty much all stack functionalities, so configuration changes made to the OpenThread stack could be implemented and tested with a single build.
Moreover having this functionality implemented as a separate sample seems even more unnatural - it would not even deliver a binary that flashed on the board would do anything useful.

If I understand correctly the SPM does not provide a simple mechanism to replace libraries which is the functionality we need to add here.

Reply:
> If I understand correctly the SPM does not provide a simple mechanism to replace libraries which is the functionality we need to add here.

Let's keep those two needs separated:
1) Linking application with precompiled libs or build libs from source
2) Copying new libraries to another location.


1 )
---

What i'm adressing, when I refer to SPM is 1).
So the possibility of linking OT cli with pre-compiled libs, or compiled from source should use a similar menu structure.
That is:
```
(X) Link with OT libraries
    (${NRFXLIB_DIR}/openthread/lib/cortex-m4/soft-float/v1.1)      OT library path
( ) Build from source
```
so that we have a uniform way of selecting `Build from source` / `link with lib`.
This furthermore allows end-users to point to a specific lib in case theu want to locate the own version in custom repo.
Just as Nordic tries to avoid patch of upstream Zephyr, then we must expect out users to have similar preference.

`OT library path` is of course the default NRFXLIB path as today.

2 )
---
See comment here: https://github.com/nrfconnect/sdk-nrf/pull/2655#discussion_r469160320

Other
-------

> Good question. My initial approach was to provide it as a completely separate example.
>   I was advised to to combine it with the CLI sample.
>
> It makes sense as the CLI can be used to test pretty much all stack functionalities, so configuration changes made to the  OpenThread stack could be implemented and tested with a single build.

Not sure I follow the rationale, but I'm not advocating for changing this.
I'm just wondering if such approach makes sense to our users.

I see the value in being able to compile from source, but I assume that it would also make sense for light switch to be able to compile from source.
If I understand the workflow correctly, then users would need to do following to use a custom version of OT libs in light switch:

1) Open cli sample
2) build cli with libs from source
3) copy libs
4) open light switch
5) build light switch with new libs

Reply:
I got the point now. Agree, it makes sense.

1) Actually this has already been implemented: https://github.com/nrfconnect/sdk-nrfxlib/pull/223
The difference between your description and PR mentioned above is the fact that the user is not allowed to enter libs location. If this should be improved we definitely need to work on it in a separate PR.

2) Yes, the workflow you described is exactly the way it works now.
Ticking a menuconfig checkbox `Substitute OpenThread Libraries` is quite explanatory, but having it in the menuconfig may still be misleading.

I will try to implement `ninja install_openthread_libraries` for all OpenThread samples. I agree this is the cleanest solution.